phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -2531,7 +2529,7 @@ public boolean execute(String sql) throws SQLException {
             }
             executeMutation(stmt, createAuditQueryLogger(stmt, sql));
             flushIfNecessary();
-            return false;
+            return isResultSetExpected(stmt);
```

Comment:
Is it fine to return `true` even if the ResultSet is null?

Reply:
I already verified that if result set is expected, it will result in non-null empty RS.